uber/models/attendee.py

```diff
@@ -826,33 +833,74 @@ def hour_map(self):
                 all_hours[hour] = shift.job
         return all_hours
 
+    @property
+    def job_visibility(self):
+        from uber.models.department import Job
+        return Job.ALL_STAFF if self.badge_type == c.STAFF_BADGE \
+            else Job.ALL_VOLUNTEERS
+
+    @cached_property
+    def available_job_filters(self):
+        from uber.models.department import Job
+
+        job_filters = [Job.is_unfilled]
+        if c.AT_THE_CON:
+            return job_filters
+
+        member_dept_ids = set(d.department_id for d in self.dept_memberships)
+        requested_dept_ids = set(
+            d.department_id for d in self.dept_membership_requests) \
+            .difference(member_dept_ids)
+
+        member_filter = Job.department_id.in_(member_dept_ids) \
+            if member_dept_ids else None
+
+        max_depts = c.MAX_DEPTS_WHERE_WORKING
+        no_max = max_depts < 1
+
+        requested_filter = None
+        if requested_dept_ids and (len(member_dept_ids) < max_depts or no_max):
+            depts_where_working = set(j.department_id for j in self.jobs)
+            if len(depts_where_working) >= max_depts and not no_max:
```

Comment:
Another nitpick -- I think it might be more readable if you avoided the double negative here ("not no_max") and just checked for truthiness in max_depts (and max_depts).

Reply:
See above. We must consider the case where `c.MAX_DEPTS_WHERE_WORKING` is less than zero (a negative integer).
```
>>> bool(-1)
True
```